lang/cs/Org.Apache.REEF.Network/StreamingCodec/Impl/DataCodec.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.IO;
+using Org.Apache.REEF.Wake.Remote;
+
+namespace Org.Apache.REEF.Network.StreamingCodec.Impl
+{
+    /// <summary>This is the abstract Codec class that encodes and decodes the type T. 
+    /// This class is mainly built for Group Communication. This class effectively inherits 
+    /// member function of ICodec<T> as abstract functions which are then used to implement 
+    /// the memebers of IDataCodecObj interface</summary>
+    public abstract class DataCodec<T> : IDataCodecObj, ICodec<T>
```

Comment:
This class is only used *within* the guts of the group communications code, right? In that case, why not make it a wrapper around a `Codec` or `StreamCodec`? That way we don't ask users to do anything different from the usual implementation of those interfaces.